// md-editor-v3/lib/es/NormalToolbar.mjs
import { defineComponent as r, createVNode as u } from "vue";
import { p as d } from "./chunks/config.mjs";
import { g as l } from "./chunks/vue-tsx.mjs";
const f = {
  title: {
    type: String,
    default: ""
  },
  /**
   * 展示在工具栏的内容，通常是个图标
   *
   * @deprecated 使用默认插槽代替
   */
  trigger: {
    type: [String, Object],
    default: void 0
  },
  onClick: {
    type: Function,
    default: void 0
  },
  /**
   * ==没有意义，仅用于规避克隆组件自动嵌入insert方法时，传入的是该组件而产生的waring
   */
  insert: {
    type: Function,
    default: void 0
  },
  language: {
    type: String,
    default: void 0
  },
  theme: {
    type: String,
    default: void 0
  },
  previewTheme: {
    type: String,
    default: void 0
  },
  codeTheme: {
    type: String,
    default: void 0
  },
  disabled: {
    type: Boolean,
    default: void 0
  },
  showToolbarName: {
    type: Boolean,
    default: void 0
  }
  /**
   * ==结束
   */
}, n = /* @__PURE__ */ r({
  name: "NormalToolbar",
  props: f,
  emits: ["onClick"],
  setup(e, t) {
    return () => {
      const o = l({
        props: e,
        ctx: t
      }, "trigger"), a = l({
        props: e,
        ctx: t
      });
      return u("button", {
        class: [`${d}-toolbar-item`, e.disabled && `${d}-disabled`],
        title: e.title,
        disabled: e.disabled,
        onClick: (i) => {
          e.onClick?.(i), t.emit("onClick", i);
        },
        type: "button"
      }, [a || o]);
    };
  }
});
n.install = (e) => (e.component(n.name, n), e);
export {
  n as default
};
